test/close.test.js

```diff
@@ -233,3 +233,37 @@ t.test('Current opened connection should not accept new incoming connections', t
     })
   })
 })
+
+test('Cannot be reopened the closed server without listen callback', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  fastify.listen(0)
+    .then(() => {
+      fastify.close()
+        .then(() => {
+          fastify.listen(0).catch(err => {
+            t.ok(err)
+            t.is(err.code, 'FST_ERR_REOPENED_CLOSE_SERVER')
+          })
+        })
+    })
+})
+
+test('Cannot be reopened the closed server has listen callback', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    fastify.close((err) => {
+      t.error(err)
+
+      fastify.listen(0, err => {
+        t.is(err.code, 'FST_ERR_REOPENED_CLOSE_SERVER')
+        t.ok(err)
+      })
+    })
```

Comment:
would you mind to use async/await in the test?

Reply:
thanks @mcollina. 
I try use async/await, while when I change below it,  show error for it `tapError: incorrect number of tests`,  plan test is 2, I dig it for a while, but not solved. please told me where is errors. 
```

test('Cannot be reopened the closed server has listen callback', async t => {
  t.plan(2)
  const fastify = Fastify()

  await fastify.listen(0)
  await fastify.close()

  fastify.listen(0, (err) => {
    t.is(err.code, 'FST_ERR_REOPENED_CLOSE_SERVER')
    t.ok(err)
  })
})
```

Reply:
you must write

```
try{
  fastify.listen(0)
}catch(err){
    t.is(err.code, 'FST_ERR_REOPENED_CLOSE_SERVER')
    t.ok(err)
}
```

Reply:
Hey @Eomm thanks for the answer.

The test unit is `listen has callback `.  if I follow  the code about 👆，I don't test code logic. https://github.com/fastify/fastify/pull/2415/files#diff-c945a46d13b34fcaff544d966cffcabaR138.

Reply:
Ops 😅

Then you can just wrap the callback in a new Promise 👍🏼

Reply:
thanks @Eomm, I got it